src/libraries/Microsoft.PowerFx.Core/Public/Types/TableType.cs

```diff
@@ -31,10 +32,20 @@ public override void Visit(ITypeVistor vistor)
             vistor.Visit(this);
         }
 
-        public TableType Add(NamedFormulaType field)
+        public TableType Add(NamedFormulaType field, string optionalDisplayName = null)
```

Comment:
>string optionalDisplayName = null [](http://example.com/codeflow?start=53&length=33)

to consider: should we add this param as part of NamedFormulaType?  They're conceptually locked, and that'd make the signatures cleaner.